### ISSUE TITLE
Accessibility enforcement and testing

### ISSUE THREAD

**Issue by david-crespo** [oxidecomputer/console#443]
TypeScript is pretty limited when it comes to enforcing correct use of accessibility attributes. For example, there's no nice way to enforce that exactly one (or even at least one) of `aria-label` or `aria-labelledby` is present. On top of that, we often won't even have a single component responsible for coordinating between controls and their labels ([example](https://github.com/oxidecomputer/console/commit/4a9f89bb1079)).

There are a number of approaches to ensuring we do a good job with accessibility, and we should do most or all of them:

* Manually test the app with a screen reader ([VoiceOver manual](https://www.apple.com/voiceover/info/guide/_1124.html))
* Assert invariants at runtime that throw errors only during development
  * Should be turned off in production, or even better, have Rollup minify them out of the bundle altogether by wrapping them in `if (process.env.NODE_ENV === 'production')`. If this `if` is inside the `invariant` function, the invariant call itself will still be made but it'll be empty. Would be cool to see if there's a clean way to take the whole call out without manually writing the `if` everywhere.
* Use tools like [axe DevTools](https://chrome.google.com/webstore/detail/axe-devtools-web-accessib/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US) or [ASLint](https://aslint.org/) to manually inspect the app
* Integrate accessibility assertions like [`jest-axe`](https://github.com/nickcolley/jest-axe) into our automated tests (but remember these are limited in what they can catch)
